internal/commands/push/command_test.go

```diff
@@ -1029,11 +1029,170 @@ Try instead: realm-cli push --local testdata/dependencies --remote appID --inclu
 	})
 }
 
+func TestPushHandlerUseCreateNewApp(t *testing.T) {
+	groupID := "groupID"
+	appID := "eggcorn-abcde"
+
+	realmClient := mock.RealmClient{}
+	realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+		return []realm.App{{GroupID: groupID, ClientAppID: appID}}, nil
+	}
+	realmClient.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+		return realm.App{
+			GroupID:     groupID,
+			ClientAppID: "eggcorn-abcde",
+			ID:          appID,
+			Name:        name,
+			AppMeta:     meta,
+		}, nil
+	}
+	realmClient.DiffFn = func(groupID, appID string, appData interface{}) ([]string, error) {
+		return []string{}, nil
+	}
+
+	t.Run("with a realm_config.json", func(t *testing.T) {
+		tmpDir, teardown, tmpDirErr := u.NewTempDir("push_handler")
+		assert.Nil(t, tmpDirErr)
+		defer teardown()
+
+		app := local.App{RootDir: tmpDir, Config: local.FileRealmConfig, AppData: &local.AppRealmConfigJSON{local.AppDataV2{local.AppStructureV2{
+			ConfigVersion:   realm.AppConfigVersion20210101,
+			Name:            "eggcorn",
+			Location:        realm.Location("location"),
+			DeploymentModel: realm.DeploymentModel("deployment_model"),
+			Environment:     realm.Environment("environment"),
+		}}}}
+
+		app.WriteConfig()
+
+		out := new(bytes.Buffer)
+		console, ui, consoleErr := mock.NewConsoleWithOptions(mock.UIOptions{AutoConfirm: true}, out)
+		assert.Nil(t, consoleErr)
+		defer console.Close()
+
+		// Run handler method on both root directory and nested directory.
+		for _, localPath := range []string{tmpDir, filepath.Join(tmpDir, "nested_directory")} {
```

Comment:
instead of this sort of for loop, can we make this a test table with proper descriptions?

```
for _, tc := range []struct {
	description     string
	localPath          string
}{
	{
		description: "....",
		localPath:    tmpDir,
	},
	{
		description:       "...",
		localPath:    filepath.Join(tmpDir, "nested_directory"),
	},
},
```

Reply:
on second thought, I don't think it's necessary for us to test creating a new app from the root dir. (you can remove that for loop on line 1074)

But we could condense all three of these cases ("with a realm_config.json", "with a config.json", and "with a stitch.json") into a test table 

```
for _, tc := range []struct {
	description          string
	configVersion        realm.AppConfigVersion
        configFile           File
        configData           string 
}{
	...
},
```